assets/chef-recipes/cookbooks/mariadb/recipes/mdberepos.rb

```diff
@@ -1,65 +1,56 @@
 include_recipe 'packages::configure_apt'
 
-# install default packages
-[ "net-tools", "psmisc" ].each do |pkg|
-  package pkg
+# Install default packages
+%w[net-tools psmisc].each do |pkg|
+  package pkg do
+    retries 2
+    retry_delay 10
+  end
 end
 
-node.set_unless['maria']['version'] = "10.0"
+# MDBE package attributes
+system "echo MDBE version: #{node['mariadb']['version']}"
+system "echo MDBE repo: #{node['mariadb']['repo']}"
+system "echo MDBE repo key: #{node['mariadb']['repo_key']}"
+
+repo_file_name = node['mariadb']['repo_file_name']
 
+# MDBE repos
 case node[:platform_family]
-when "debian"
-  # Add repo key
-  execute "Key add" do
-    command "apt-key adv --recv-keys --keyserver keyserver.ubuntu.com 0xce1a3dd5e3c94f49"
-  end
-  release_name = '$(lsb_release -cs)'
-  # Add repo
-  execute "Repository add" do
-    command 'echo "deb http://code.mariadb.com/mariadb-enterprise/'+ node['maria']['version'] + '/repo/' + node[:platform] + ' ' + release_name + ' main" > /etc/apt/sources.list.d/mariadb.list'
-  end
-  execute "update" do
-    command "apt-get update"
-  end
-when "rhel", "fedora"
+when 'debian', 'ubuntu'
+  # Split MaxScale repository information into parts
+  repository = node['mariadb']['repo'].split(/\s+/)
+  apt_repository 'mariadb' do
```

Comment:
This resource contains sensitive data. Please provide such option for the resource.